bench/bm_history_buffer.cpp

```diff
@@ -0,0 +1,150 @@
+#include "benchmark.hpp"
+
+#include <boost/ut.hpp>
+#include <chrono>
+#include <iostream>
+#include <thread>
+
+#include <fmt/format.h>
+
+#include <circular_buffer.hpp>
+#include <history_buffer.hpp>
+#include <utils.hpp>
+
+inline const boost::ut::suite _buffer_tests = [] {
+    constexpr uint32_t n_repetitions = 10;
+    constexpr uint32_t samples       = 10'000'000; // minimum number of samples
+    using namespace benchmark;
+    using namespace gr;
+
+    {
+        circular_buffer<int, std::dynamic_extent, ProducerType::Multi> buffer(32);
+        auto                                                           writer                   = buffer.new_writer();
+        auto                                                           reader                   = buffer.new_reader();
+
+        "circular_buffer<int>(32) - multiple producer"_benchmark.repeat<n_repetitions>(samples) = [&writer, &reader] {
```

Comment:
How is this test a multiple producer - should the example have multiple writers?

Reply:
The multi-producer case is tested already in another unit-test and benchmark. Here it's just testing the multi-producer strategy which should -- in theory -- be a bit more computational intensive. Actually, it turns out -- maybe this is a simplified testing artefact -- that the circular_buffer is faster than the history_buffer. Need to investigate why/how the compiler can optimise one better than the other (the history_buffer is basically simpler and should be thus faster)... :thinking: